src/lucky_record/queryable.cr

```diff
@@ -50,11 +50,23 @@ module LuckyRecord::Queryable(T)
     exec_query.first? || raise RecordNotFoundError.new(model: @@table_name, query: :first)
   end
 
+  def first?
+    first
+  rescue RecordNotFoundError
```

Comment:
Raising and rescuing is a fairly expensive operation, so it's better to avoid it if possible. So instead I think this would be more performant if it looked a bit more like the issue, where `first?` and `last?` did something like this:

```crystal
# Basically the existing `first` method, but it doesn't raise
def first?
  query.limit(1)
  exec_query.first?
end	

def first
  first? || raise RecordNotFoundError.new(model: @@table_name, id: id.to_s)
end	      
```

This way we don't raise/rescue unnecessarily

Reply:
Okay I wasn't sure about the performance penalty, this clears it up thanks

Reply:
I guess I didn't read the issue properly because I completely missed the suggested implementation